game-core/src/main/java/games/strategy/engine/data/RouteFinder.java

```diff
@@ -12,18 +13,32 @@
 // TODO this class doesn't take movementcost into account... typically the shortest route is the fastest route, but not
 // always...
 class RouteFinder {
+
   private final GameMap map;
   private final Predicate<Territory> condition;
   private final Map<Territory, Territory> previous;
+  private final Collection<Unit> units;
+  private final PlayerID player;
 
   RouteFinder(final GameMap map, final Predicate<Territory> condition) {
     this.map = map;
     this.condition = condition;
     previous = new HashMap<>();
+    units = new HashSet<>();
+    player = null;
+  }
+
+  RouteFinder(final GameMap map, final Predicate<Territory> condition, final Collection<Unit> units,
+      final PlayerID player) {
+    this.map = map;
+    this.condition = condition;
+    previous = new HashMap<>();
+    this.units = units;
+    this.player = player;
```

Comment:
I don't particularly like how this class works but decided not to make major refactoring changes here.